### PR TITLE
Fix template restart bug

### PR DESCRIPTION
**Pull request by LebronAl**
In the current implementation, the template-based time series is restored by calling the getMNodes function after a restart. However, the getMNodes function currently does not support template lookup and this needs to be fixed.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 1]
```diff
@@ -804,10 +804,12 @@ public TSDataType getSeriesType(PartialPath path) throws MetadataException {
...
   public MeasurementMNode[] getMNodes(PartialPath deviceId, String[] measurements)
       throws MetadataException {
-    MNode deviceNode = getNodeByPath(deviceId);
     MeasurementMNode[] mNodes = new MeasurementMNode[measurements.length];
     for (int i = 0; i < mNodes.length; i++) {
-      mNodes[i] = ((MeasurementMNode) deviceNode.getChild(measurements[i]));
+      try {
+        mNodes[i] = (MeasurementMNode) getNodeByPath(deviceId.concatNode(measurements[i]));
+      } catch (PathNotExistException ignored) {
+      }
```
Comment (wangchao316) [inline server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:812]: hi,  I have a minor..
If an exception occurs, a line of logs must be recorded to facilitate fault tracing.

Comment (LebronAl) [reply server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:813]: OK

**Review by HTHou — APPROVED** [review]
lgtm

**Review by wangchao316 — APPROVED** [review]
LGTM

**Comment by LebronAl** [comment]
> LGTM

Hi, it seems that the failure of E2E test has nothing to do with this PR, but with [PR 3292](https://github.com/apache/iotdb/pull/3292). Could you please check what the problem is?
image

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**